### DISCUSSION TITLE
Random() Functionality in programming languages

### DISCUSSION THREAD

**Discussion by baderouaich** [community/community#21384] [Other Feature Feedback, Questions, & Ideas]
<p>Almost every programmer have used the <strong>Random()</strong> Functionality in any programming language they use.</p>
<p><strong>but</strong> , have you ever asked yourself how this functionality works ? to be more specific.</p>
<pre><code>How the Random() Functionality is generating a random object even if nothing is actually random inside a computer?
</code></pre>
<p>Well, hopefully someone has an idea about that. thanks in advance.</p>

**Comment by pierDipi** [comment, marked as answer]
<p>That’s a very good question!</p>
<p>Behind pseudo-random functions there are math and statistics’ theorems; I think the following wiki page, <a href="https://en.wikipedia.org/wiki/Linear_congruential_generator" rel="nofollow noopener">Linear congruential generator</a>, is a good starting point to understand a little bit of this huge topic.</p>

**Comment by baderouaich** [comment]
<p><a class="mention" href="/pierdipi">@pierdipi</a>  Thank you for clarifying! looking forward to learn more in this huge topic.</p>
